core-test/src/test/scala/org/bitcoins/core/util/CryptoUtilTest.scala

```diff
@@ -55,4 +59,14 @@ class CryptoUtilTest extends FlatSpec with MustMatchers {
     CryptoUtil.sha256Hash160(hex).hex must be(expected)
     CryptoUtil.sha256Hash160(hex).flip.flip.hex must be(expected)
   }
-}
+
+  it must "recover the two public keys used to sign a message" in {
+    PropertyChecks.forAll(CryptoGenerators.privateKey, CryptoGenerators.sha256Digest) {
+      case (privKey, hash) =>
```

Comment:
What does the `case` keyword mean here?

Reply:
The `case` keyword used to enumerate a condition that your code should handle. 

You can use case inside of pattern matches like 

```scala
val x = "Hello World" 
x match {
  case "Hello World" => println("matched found hello world") 
  case _: String => println("Did not match hello world")
}
```
You can also use `case` inside of lambdas like this 

```scala
//define an anonymous function that prints the given string
{ case x : String => println(x) }
```
but you could also write this

```scala
{ x: String => println(x) }
```

In the specific instance you are talking about, the `case` keyword is not strictly necessary, and I could have removed it. One of the good things (and bad things) about Scala is you can be extremely flexible with syntax.